modules/packages/Sort.chpl

```diff
@@ -1617,6 +1690,44 @@ module ShallowCopy {
 
   // TODO: move these out of the Sort module and/or consider
   // language support for it.
+  inline proc shallowCopy(ref dst:?t, ref src:t) {
+    if isPODType(t) {
+      dst = src;
+    } else {
+      var size = c_sizeof(t);
+      c_memcpy(c_ptrTo(dst), c_ptrTo(src), size);
+      // The version moved from should never be used again,
+      // but we clear it out just in case.
+      c_memset(c_ptrTo(src), 0, size);
+    }
+  }
+
+  // returns the result of shallow copying src
+  pragma "unsafe"
+  inline proc shallowCopyInit(ref src:?t) {
+    var dst: t;
+    shallowCopy(dst, src);
+    return dst;
+  }
+
+  pragma "unsafe"
+  inline proc shallowSwap(ref lhs:?t, ref rhs:t) {
+    pragma "no auto destroy"
+    var tmp: t;
+    if isPODType(t) {
+      tmp = lhs;
+      lhs = rhs;
+      rhs = tmp;
```

Comment:
OOC, why isn't this `lhs <=> rhs;`?

Reply:
I wanted it to match the memcpy version a bit